src/utilities/plugin-utils.test.ts

```diff
@@ -0,0 +1,200 @@
+import { Activity, Plugin } from "../types";
+import { addUsedPlugin, clearUsedPlugins, findUsedPlugins, getGlossaryEmbeddable, getUsedPlugins, IEmbeddablePluginContext, initializePlugin, IPartialEmbeddablePluginContext, loadLearnerPluginState, loadPluginScripts, validateEmbeddablePluginContextForPlugin, validateEmbeddablePluginContextForWrappedEmbeddable } from "./plugin-utils";
+import sampleActivityGlossaryPlugin from "../data/sample-activity-glossary-plugin.json";
+import { LaraGlobalType } from "../lara-plugin";
+
+describe("Plugin utility functions", () => {
+
+  const plugin: Plugin = {
+    description: "Test Plugin",
+    author_data: "",
+    approved_script_label: "test",
+    component_label: "Test",
+    approved_script: {
+      name: "Test Script",
+      url: "https://example.com/plugin.js",
+      label: "test",
+      description: "Test Script",
+      version: "1.0.0",
+      json_url: "https://example.com/manifest.json",
+      authoring_metadata: ""
+    }
+  };
+
+  const activity: Activity = sampleActivityGlossaryPlugin as Activity;
+
+  const partialContext: IPartialEmbeddablePluginContext = {
+    LARA: {} as any,
+    embeddable: {} as any,
+    embeddableContainer: {} as any,
+    wrappedEmbeddable: {} as any,
+    wrappedEmbeddableContainer: {} as any
+  };
+
+  describe("#addUsedPlugin", () => {
+    beforeEach(() => clearUsedPlugins());
+
+    it("adds a used plugin when not already in the list", () => {
+      expect(getUsedPlugins()).toEqual([]);
+      addUsedPlugin(plugin);
+      expect(getUsedPlugins()).toEqual([{
+        id: 0,
+        loaded: false,
+        plugin
+      }]);
+    });
+
+    it("does not add a used plugin when it is already in the list", () => {
+      expect(getUsedPlugins().length).toEqual(0);
+      addUsedPlugin(plugin);
+      expect(getUsedPlugins().length).toEqual(1);
+      addUsedPlugin(plugin);
+      expect(getUsedPlugins().length).toEqual(1);
+    });
+  });
+
+  describe("#findUsedPlugins", () => {
+    it("works in teacher edition mode", () => {
+      const usedPlugins = findUsedPlugins(activity, true);
+      expect(usedPlugins.length).toBe(3);
```

Comment:
Since the usedPlugins length is the same with and without teacher edition mode, does this mean that the test activity doesn't have any teacher edition plugins?

Could you update this to use an activity with teacher edition plugins so it would exercise more of the code?

Reply:
This has been fixed in this commit (https://github.com/concord-consortium/activity-player/pull/132/commits/4b9eaad5d731de51119f9ce09c85894f2d9bc565).  I needed to clear the used plugins cache between tests so that accurate counts could be tested.  I also changed from just counting the plugins to comparing the plugin script labels.